polaris-core/src/main/java/org/apache/polaris/core/entity/TableLikeEntity.java

```diff
@@ -29,6 +29,14 @@ public class TableLikeEntity extends PolarisEntity {
   // of the internalProperties JSON file.
   public static final String METADATA_LOCATION_KEY = "metadata-location";
 
+  // For applicable types, this key on the "internalProperties" map will return the content of the
+  // metadata.json file located at `METADATA_CACHE_LOCATION_KEY`
+  private static final String METADATA_CACHE_CONTENT_KEY = "metadata-cache-content";
+
+  // For applicable types, this key on the "internalProperties" map will return the location of the
+  // `metadata.json` that is cached in METADATA_CACHE_CONTENT_KEY
+  private static final String METADATA_CACHE_LOCATION_KEY = "metadata-cache-location";
```

Comment:
Since we're updating TableLikeEntity atomically, it shouldn't be possible for us to ever cache mismatched content vs the base `metadata-location` key. If we ever do find ourselves relying on a possible mismatch between `metadata-location` and `metadata-cache-location`, it's probably a crutch pointing at a deeper concurrency issue.